build/common.sh

```diff
@@ -1427,3 +1437,40 @@ function kube::release::gcs::publish() {
     return 1
   fi
 }
+
+# ---------------------------------------------------------------------------
+# Docker Release
+
+# Releases all docker images to a docker registry specified by KUBE_DOCKER_REGISTRY
+# using tag KUBE_DOCKER_IMAGE_TAG.
+#
+# Globals:
+#   KUBE_DOCKER_REGISTRY
+#   KUBE_DOCKER_IMAGE_TAG
+# Returns:
+#   If new pushing docker images was successful.
+function kube::release::docker::release() {
+  local binaries=(
+    "kube-apiserver"
+    "kube-controller-manager"
+    "kube-scheduler"
+    "kube-proxy"
+  )
+
+  local platforms=(
+    "linux-amd64"
```

Comment:
AFAIK @brendandburns has pushed the pause, skydns, exechealthz and other addon images as experimental images. 

The thing is just that we have to keep it consistent. I am working on first-class ARM support and the naming here is quite important.

I think we should go with arch only naming, because docker runs natively only on linux for the time being. If we want to support darwin or windows in the future is another (probably long) discussion